storage/remote/client.go

```diff
@@ -148,10 +153,25 @@ func NewWriteClient(name string, conf *ClientConfig) (WriteClient, error) {
 		Client:           httpClient,
 		retryOnRateLimit: conf.RetryOnRateLimit,
 		timeout:          time.Duration(conf.Timeout),
-		headers:          conf.Headers,
 	}, nil
 }
 
+func newRoundTripperWithHeaders(h map[string]string, underlyingRT http.RoundTripper) *roundTripperWithHeaders {
+	return &roundTripperWithHeaders{headers: h, RoundTripper: underlyingRT}
+}
+
+type roundTripperWithHeaders struct {
```

Comment:
Naming is hard, but I think this does not tell what this RT is doing 🤗 

What about this:
```suggestion
type injectHeadersRoundTripper struct {
```
?

Reply:
Sorry, I felt that `..WithHeaders` justified that the round-tripper being implemented, has headers along the way (as newRoundTripperWithHeaders implements headers on to the underlying round-tripper).

Still, if we want `injectHeadersRoundTripper`, then I guess it would be `injectHeadersInRoundTripper`, which probably makes it too explanatory. WDYT? @bwplotka

Reply:
injectHeadersRoundTripper is fine. It is not "in" roundtripper because in itlself it is a roundtripper.